src/transcoding.ts

```diff
@@ -248,16 +249,33 @@ export function requestChangeCaseAndCleanup(
     if (!Object.prototype.hasOwnProperty.call(request, field)) {
       continue;
     }
-    const convertedField = caseChangeFunc(field);
+    let convertedField = caseChangeFunc(field);
+
+    // Here, we want to check if the fields in the proto match
+    // the fields we are changing; if not, we assume it's user
+    // input and revert back to its original form
+    if (
+      fieldsToChange &&
+      fieldsToChange?.size !== 0 &&
```

Comment:
The size check is technically not needed, `.has()` is a no-op for an empty set.

Reply:
This was breaking tests, because it would enter this comparison if there was an empty set, and, since it would find no match, this would evaluate to true, instead of false: `!fieldsToChange?.has(convertedField)`

thus reverting convertedField back to field.